content/frontend/react-apollo/9-pagination.md

```diff
@@ -1,68 +1,115 @@
+  exposes the required arguments for lists.'
+question:
+  "What's the difference between the 'query' and 'readQuery'
+  methods on the 'ApolloClient'?"
+answers:
+  [
+    "'readQuery' always fetches data over the network while
+    'query' can retrieve data either from the cache or
+    remotely",
+    "'readQuery' can only be used to reading data while
+    'query' can also be used to write data",
+    "'readQuery' was formerly called 'query' and the
+    functionality of both is identical",
+    "'readQuery' always reads data from the local cache
+    while 'query' might retrieve data either from the cache
+    or remotely"
+  ]
 correctAnswer: 3
-videoId: ""
-duration: 0		
-videoAuthor: ""
+videoId: ''
+duration: 0
+videoAuthor: ''
 ---
 
-The last topic that we'll cover in this tutorial is pagination. You'll implement a simple pagination approach so that users are able to view the links in smaller chunks rather than having an extremely long list of `Link` elements.
+The last topic that we'll cover in this tutorial is
+pagination. We'll implement a simple pagination approach so
+that users are able to view the links in smaller chunks
+rather than having an extremely long list of `Link`
+elements.
 
 ## Preparing the React Components
 
-Once more, you first need to prepare the React components for this new functionality. In fact, we'll slightly adjust the current routing setup. Here's the idea: The `LinkList` component will be used for two different use cases (and routes). The first one is to display the 10 top voted links. Its second use case is to display new links in a list separated into multiple pages that the user can navigate through.
+Once more, we fir needst to prepare the React components for
+this new functionality. In fact, we'll make a slight
+adjustment to the current routing setup. The idea is that
+the `LinkList` component will be used for two different
+purposes (and routes). The first one is to display the top
+ten voted links and the second use case is to display new
+links in a list separated into multiple pages that the user
+can navigate through.
 
 <Instruction>
 
-Open `App.js` and adjust the render method like so:
-
-```js{7,11,12}(path=".../hackernews-react-apollo/src/components/App.js")
-render() {
-  return (
-    <div className='center w85'>
-      <Header />
-      <div className='ph3 pv1 background-gray'>
-        <Switch>
-          <Route exact path='/' render={() => <Redirect to='/new/1' />} />
-          <Route exact path='/create' component={CreateLink} />
-          <Route exact path='/login' component={Login} />
-          <Route exact path='/search' component={Search} />
-          <Route exact path='/top' component={LinkList} />
-          <Route exact path='/new/:page' component={LinkList} />
-        </Switch>
-      </div>
+Open `App.js` and adjust the component to look like this:
+
+```js{}(path=".../hackernews-react-apollo/src/components/App.js")
```

Comment:
Kindly provide a diff to show the user what is changing in this file.
```suggestion
```js{20-24}(path=".../hackernews-react-apollo/src/components/App.js") 
```